### DISCUSSION TITLE
Viewport prefetching - Prefetching the same URL multiple times

### DISCUSSION THREAD

**Discussion by tervay** [remix-run/remix#9677] [Q&A] [answered]
I've found that scrolling a `<Link prefetch="viewport" ...` component will prefetch the target URL multiple times. I'm not sure if this is a bug, feature, or undefined behavior. I couldn't find any mention of it in the docs. Semi-related, I'm also not super sure why so many prefetches get canceled. I'm aware I'm probably being a bit much with the number of prefetches here, but it's partially due to the links prefetching multiple times.

https://github.com/remix-run/remix/assets/7595639/40d58fec-117c-4520-a7c5-4be5b7f50ad8

Is there any recommended way around this? Is this intentional?

Thanks.

**Comment by sergiodxa** [comment, marked as answer]
Both are intended behaviors.

The prefetch in viewport is triggered again to ensure that if the user saw link to /a and half an hour later see it again it will prefetch it again instead of relying on old prefetched data or not using any prefetch.

The cancelled request is handled by the browser when the link tags to prefetch the route are removed from the DOM once the user scroll past the links and the link tags are removed.
